scio-core/src/main/scala/com/spotify/scio/coders/EmptyOnSerializeAtomicReference.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.coders
+
+import java.io.{
+  IOException,
+  ObjectInputStream,
+  ObjectOutputStream,
+  ObjectStreamException
+}
+import java.util.concurrent.atomic.AtomicReference
+
+private abstract class EmptyOnSerializeAtomicReference[T] extends Serializable {
```

Comment:
Could you pls add comments here why do we need this?

Reply:
@ravwojdyla @bilisie I think we can remove this by just using `@transient lazy val`

Reply:
@regadas @bilisie See comment in the new round below.